File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int i=0,j=heights.size()-1;
        int out=0;
        while(i<j){
            //cout<<i<<" "<<j<<" "<<heights[j]<<" "<<heights[i]<<" "<<min(heights[j],heights[i])<<endl;
            out=max(out, min(heights[j],heights[i])*(j-i));
            if(heights[j]>heights[i]) i++;
            else if(heights[j]<=heights[i]) j--;
        }
        return out;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        location_map = defaultdict(int)
        longest_seq = 0
        last_index = -1
        for idx, char in enumerate(s):
            if char in location_map:
                last_index = max(last_index, location_map[char])
                
            longest_seq = max(longest_seq, idx - last_index)
            location_map[char] = idx

        return longest_seq

        